contracts/connectors/loantoken/LoanTokenLogicStandard.sol

```diff
@@ -1382,6 +1398,27 @@ contract LoanTokenLogicStandard is LoanTokenSettingsLowerAdmin {
 		require(!isPaused, "unauthorized");
 	}
 
+	/**
+	 * @notice Make sure caller did not perform any previous operation in the
+	 *   current block.
+	 *
+	 * @dev To protect from the lending fees manipulation using flash loan we
+	 * need to prevent lending and withdrawing from the pools in the same
+	 * tx/block by the same account.
+	 * */
+	function _checkNotInTheSameBlock() internal {
+		uint256 _currentBlock;
+
+		/// @dev Get and buffer current block.
+		_currentBlock = block.number;
+
+		/// @dev Check there are no previous txs in this block coming from same account.
+		require(!hasInteracted[_currentBlock][msg.sender], "Avoiding flash loan attack: several txs in same block from same account.");
```

Comment:
checking against `msg.sender` doesn't protect against FL, only complicates the setup:
the attacker would need to add extra contracts A1, A2, ... for each individual interaction within their transaction 

formally, you can achieve FL protection via `tx.origin` check, however it is discouraged.